README.md
# Onmyoji_Python


大风刮倒梧桐树，自有旁人论长短


## 主要功能
1. 御魂副本
2. 组队永生之海副本
3. 业原火副本
4. 御灵副本
5. 个人突破
6. 寮突破
7. 道馆突破
8. 普通召唤
9. 百鬼夜行
10. 限时活动
11. 组队日轮副本


## 功能说明
1. 御魂副本
   - 组队/单人
   - 组队司机/打手
   - 十层/悲鸣/神罚
2. 组队永生之海副本
    - 仅适用于组队中
    - 适配司机/打手
3. 业原火副本
4. 御灵副本
5. 个人突破
    - 3胜刷新
    - 锁定阵容
6. 寮突破
    - 锁定阵容，从上至下进攻
7. 道馆突破
    - 等待系统进入/手动挑战/正在进行中
    - 挂机阵容
8. 普通召唤
    - 清票/狗粮
9. 百鬼夜行
   - 清票
   - 截图，默认启用
10. 限时活动
11. 组队日轮副本


## 环境安装
* 阴阳师桌面版 [提供NGA下载地址](https://nga.178.com/read.php?tid=29661629)
* 本项目 [releases](https://github.com/AquamarineCyan/Onmyoji_Python/releases)


## 使用方法

~~**无论哪种都需要管理员权限**~~

1. 前往 [releases](https://github.com/AquamarineCyan/Onmyoji_Python/releases) 下载解压打包完成的应用程序，点开即用
2. 需要一定的基础，更新较勤，可能存在bug  
   - 使用`git`命令下载源码  
     `git close https://github.com/AquamarineCyan/Onmyoji_Python.git`  
     后续只需`git pull`
   - 安装依赖 
     - venv 方式 
       `pip install -r requestments.txt`
     - poetry 方式
       `poetry install`
   - 自行打包，打包配置已存在`main.spec`    
     `pyinstaller main.spec`
   - 或者不打包，直接运行（理论上能够生成UI）  
     `python main.py`


## UI

- 功能模块
    - 1.御魂副本
    - 2.组队永生之海副本
    - 3.业原火副本
    - 4.御灵副本
    - 5.个人突破
    - 6.寮突破
    - 7.道馆突破
    - 8.普通召唤
    - 9.百鬼夜行
    - 10.限时活动
    - 11.组队日轮副本
- 游戏检测
    - 手动更新游戏窗口信息，适合窗口移动，或双开
- 设置
    - 更新方式
      - ghproxy/GitHub
    - 悬赏封印
      - 接受/拒绝/忽略/关闭


## 更新记录

[CHANGELOG.MD](https://github.com/AquamarineCyan/Onmyoji_Python/blob/main/CHANGELOG.MD)


## Tips

**请自行合理使用，所产生的一切后果自负**

移动游戏窗口后，点击 `游戏检测` 即可

由于官方更新了~~很多~~较多UI，目前仅适配部分高频功能（例魂土）


CHANGELOG.MD
# Change Log

## 最近更新

### 优化

- 日志保存的时间精度
- 添加程序复用属性
- 优化配置文件
- 修改项目结构

### Bug 修复

- 活动卡顿引起的判定异常问题

## v1.7.4 - 2023-05-24

### 新功能

- 添加手动重启按钮

### 优化

- 组队/单人御魂副本任务
- 修改工具类函数的导入形式
- 更新下载显示
- 御魂副本支持素材复用
- 组队永生之海副本，支持新版结算
- 结果判断逻辑，防止手快提前进入结算逻辑
- 重启函数
- 适配活动「宴场维和」

### Bug 修复

- 关闭悬赏可能导致进程卡住问题
- 设置-更新模式未生效的问题
- 组队御魂中场景判定问题
- 业原火无法挑战的问题

## v1.7.3 - 2023-04-26

### 新功能

- 更新自动重启
- 新增`Coor`坐标类
- 添加御魂副本测试项，使用新方法识别
- 新增悬赏封印关闭选项，降低程序运行功耗（需重启）
- 添加单人御魂副本测试功能

### 优化

- 默认启用镜像源下载
- 完善资源检查
- 测试用log文件夹测试完成后自动删除
- 更新「妖塔燃战」限时活动资源
- 设置界面

### Bug 修复

- 镜像源启用失败的问题
- 匹配更新包空路径的问题
- 资源检测线程问题
- `requirements.txt` 环境问题([#3](https://github.com/AquamarineCyan/Onmyoji_Python/issues/3))
- 图像文件后缀补全失败的问题

## v1.7.2 - 2023-03-26

### 破坏性更新

- 移除`update.json`，移除旧版本更新方式

### 新功能

- 对中文路径的提示
- 日志自动清理（30天）
- 更新记录
- 对组队御魂副本时司机可能出现的bug提示
- 御魂掉落统计beta

### 优化

- 配置文件的读取优先级
- 游戏窗口检测

### Bug 修复

- 潜存在的路径转义问题
- 神罚素材不全问题
- 日志文件概率乱码的问题
- utf-8异常乱码问题

## v1.7.1 - 2023-02-19

### 新功能

- 镜像下载源
- 对部分素材的路径检查

### 优化

- 任务完成的表现形式
- 系统通知保留至通知栏
- 适配神罚副本

## v1.7.0 - 2023-01-18

### 破坏性更新

- 该版本为1.7测试版
- 资源文件夹由`pic`更新为`resource`，尝试兼容旧资源文件夹

### 新功能

- 配置文件及设置

### 优化

- 悬赏封印处理方式并添加设置项

## v1.6.8 - 2022-12-24

### 优化

- 适配签到2000天奖励「鎏金圣域」

### Bug 修复

- 匹配道馆突破场景失败的提示

## v1.6.7 - 2022-11-20

### 新功能

- 系统通知
- 创建poetry打包环境

### 优化

- 项目结构
- 部分素材
- 重构日志工具包
- 删除run.bat启动方式（已实现自动获取管理员权限）
- 去除无更新时的系统通知
- 寮突破21时无限突破的提示
- 开启环境自动检测
- 回退限时活动至周年庆版本前    

## v1.6.6 - 2022-10-16

### 破坏性更新

- 更换检查更新方式，旧版代码将于`v1.7.0`作废

### 新功能

- 自动下载更新包
- 强制缩放([#1](https://github.com/AquamarineCyan/Onmyoji_Python/issues/1))
- 强制缩放及弹窗提示

### 优化

- 素材
- 限时活动单次上线
- 窗口大小匹配提示
- 寮突破次数选择上限

## v1.6.5 - 2022-09-21

### 新功能

- 管理员运行提示

### 优化

- 周年庆活动

## v1.6.4 - 2022-07-24

### 新功能

- 悬赏封印
- 检查更新

### Bug 修复

- UI

## v1.6.3 - 2022-06-22

### 新功能

- 环境文件requirements.txt
- 清理日志
- 限时活动
- 组队日轮副本

### 优化

- 召唤功能结束
- 截图功能
- 窗口信息显示方式
- 个人突破刷新逻辑

### Bug 修复

- 因呱太入侵导致的图像错位
- 更换pathlib库引发的错误

## v1.6.2 - 2022-05-22

### 新功能

- 业原火副本
- 道馆突破观战功能

### 优化

- 功能列表顺序

### Bug 修复

- 初次使用无日志时写入报错

## v1.6.1 - 2022-04-25

### 新功能
- UI界面色彩显示
- 组队永生之海副本
- 更新窗口信息功能

### 优化

- 代码冗余

## v1.6 - 2022-04-10

### 新功能

- UI界面
- 日志模块

### 优化

- 代码逻辑

## v1.5 - 2022-03-15

### 新功能

- 百鬼夜行功能
- 道馆突破功能
- 御灵功能
- 绘卷查分功能
- 错误和异常判断处理
- 部分注释说明
- 输出界面的色彩直观表达
- 功能选择的输入稳定性

### 优化
- 普通召唤的判定逻辑
- 个人结界与寮结界的相互识别跳转
- 文件处理的逻辑
- 适配键入回车默认选择
- 适配新版结算速度

## v1.4 - 2022-02-26

### 新功能

- 个人结界功能

### 优化

- 图片素材存储

## v1.3 - 2022-02-01

### 新功能

- 寮突破功能

### 优化

- 函数调用
- 御魂组队情况

## v1.2 - 2022-01-24

### 新功能

- 组队御魂功能

## v1.1

### 新功能

- 普通召唤功能

ui/mainui.py
# -*- coding: utf-8 -*-

################################################################################
## Form generated from reading UI file 'mainui.ui'
##
## Created by: Qt User Interface Compiler version 6.5.0
##
## WARNING! All changes made in this file will be lost when recompiling UI file!
################################################################################

from PySide6.QtCore import (QCoreApplication, QDate, QDateTime, QLocale,
    QMetaObject, QObject, QPoint, QRect,
    QSize, QTime, QUrl, Qt)
from PySide6.QtGui import (QBrush, QColor, QConicalGradient, QCursor,
    QFont, QFontDatabase, QGradient, QIcon,
    QImage, QKeySequence, QLinearGradient, QPainter,
    QPalette, QPixmap, QRadialGradient, QTransform)
from PySide6.QtWidgets import (QApplication, QButtonGroup, QCheckBox, QComboBox,
    QGroupBox, QHBoxLayout, QLabel, QLineEdit,
    QMainWindow, QPushButton, QRadioButton, QScrollArea,
    QSizePolicy, QSpacerItem, QSpinBox, QStackedWidget,
    QTabWidget, QTextBrowser, QWidget)

class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        if not MainWindow.objectName():
            MainWindow.setObjectName(u"MainWindow")
        MainWindow.resize(550, 450)
        self.centralwidget = QWidget(MainWindow)
        self.centralwidget.setObjectName(u"centralwidget")
        self.tabWidget = QTabWidget(self.centralwidget)
        self.tabWidget.setObjectName(u"tabWidget")
        self.tabWidget.setGeometry(QRect(0, 0, 551, 451))
        self.tab_home = QWidget()
        self.tab_home.setObjectName(u"tab_home")
        self.groupBox_basic = QGroupBox(self.tab_home)
        self.groupBox_basic.setObjectName(u"groupBox_basic")
        self.groupBox_basic.setGeometry(QRect(4, 20, 231, 131))
        font = QFont()
        font.setFamilies([u"\u5b8b\u4f53"])
        self.groupBox_basic.setFont(font)
        self.button_game_handle = QPushButton(self.groupBox_basic)
        self.button_game_handle.setObjectName(u"button_game_handle")
        self.button_game_handle.setGeometry(QRect(70, 90, 71, 23))
        font1 = QFont()
        font1.setFamilies([u"\u5b8b\u4f53"])
        font1.setPointSize(10)
        self.button_game_handle.setFont(font1)
        self.combo_choice = QComboBox(self.groupBox_basic)
        self.combo_choice.setObjectName(u"combo_choice")
        self.combo_choice.setGeometry(QRect(81, 24, 141, 19))
        font2 = QFont()
        font2.setFamilies([u"\u5b8b\u4f53"])
        font2.setPointSize(9)
        self.combo_choice.setFont(font2)
        self.label_function = QLabel(self.groupBox_basic)
        self.label_function.setObjectName(u"label_function")
        self.label_function.setGeometry(QRect(26, 27, 26, 15))
        self.label_function.setFont(font1)
        self.spinB_num = QSpinBox(self.groupBox_basic)
        self.spinB_num.setObjectName(u"spinB_num")
        self.spinB_num.setGeometry(QRect(81, 55, 51, 19))
        self.spinB_num.setFont(font1)
        self.spinB_num.setMaximum(999)
        self.label_numbers = QLabel(self.groupBox_basic)
        self.label_numbers.setObjectName(u"label_numbers")
        self.label_numbers.setGeometry(QRect(26, 57, 26, 15))
        self.groupBox_senior = QGroupBox(self.tab_home)
        self.groupBox_senior.setObjectName(u"groupBox_senior")
        self.groupBox_senior.setGeometry(QRect(6, 156, 231, 201))
        self.stackedWidget = QStackedWidget(self.groupBox_senior)
        self.stackedWidget.setObjectName(u"stackedWidget")
        self.stackedWidget.setGeometry(QRect(10, 20, 211, 161))
        self.none_0 = QWidget()
        self.none_0.setObjectName(u"none_0")
        self.stackedWidget.addWidget(self.none_0)
        self.yuhun_1 = QWidget()
        self.yuhun_1.setObjectName(u"yuhun_1")
        self.label_driver = QLabel(self.yuhun_1)
        self.label_driver.setObjectName(u"label_driver")
        self.label_driver.setGeometry(QRect(20, 55, 48, 21))
        font3 = QFont()
        font3.setPointSize(9)
        self.label_driver.setFont(font3)
        self.button_driver_False = QRadioButton(self.yuhun_1)
        self.buttonGroup_driver = QButtonGroup(MainWindow)
        self.buttonGroup_driver.setObjectName(u"buttonGroup_driver")
        self.buttonGroup_driver.addButton(self.button_driver_False)
        self.button_driver_False.setObjectName(u"button_driver_False")
        self.button_driver_False.setGeometry(QRect(78, 55, 38, 20))
        self.button_driver_False.setFont(font3)
        self.button_driver_False.setMouseTracking(True)
        self.button_driver_True = QRadioButton(self.yuhun_1)
        self.buttonGroup_driver.addButton(self.button_driver_True)
        self.button_driver_True.setObjectName(u"button_driver_True")
        self.button_driver_True.setGeometry(QRect(128, 55, 38, 20))
        self.button_driver_True.setFont(font3)
        self.button_passengers_3 = QRadioButton(self.yuhun_1)
        self.buttonGroup_passengers = QButtonGroup(MainWindow)
        self.buttonGroup_passengers.setObjectName(u"buttonGroup_passengers")
        self.buttonGroup_passengers.addButton(self.button_passengers_3)
        self.button_passengers_3.setObjectName(u"button_passengers_3")
        self.button_passengers_3.setGeometry(QRect(128, 81, 33, 20))
        self.button_passengers_3.setFont(font3)
        self.button_passengers_2 = QRadioButton(self.yuhun_1)
        self.buttonGroup_passengers.addButton(self.button_passengers_2)
        self.button_passengers_2.setObjectName(u"button_passengers_2")
        self.button_passengers_2.setGeometry(QRect(79, 81, 33, 20))
        self.button_passengers_2.setFont(font3)
        self.label_passengers = QLabel(self.yuhun_1)
        self.label_passengers.setObjectName(u"label_passengers")
        self.label_passengers.setGeometry(QRect(20, 81, 48, 21))
        self.label_passengers.setFont(font3)
        self.label_tips = QLabel(self.yuhun_1)
        self.label_tips.setObjectName(u"label_tips")
        self.label_tips.setGeometry(QRect(20, 130, 141, 31))
        self.button_yuhun_drop_statistics = QCheckBox(self.yuhun_1)
        self.button_yuhun_drop_statistics.setObjectName(u"button_yuhun_drop_statistics")
        self.button_yuhun_drop_statistics.setGeometry(QRect(20, 109, 100, 20))
        sizePolicy = QSizePolicy(QSizePolicy.Fixed, QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.button_yuhun_drop_statistics.sizePolicy().hasHeightForWidth())
        self.button_yuhun_drop_statistics.setSizePolicy(sizePolicy)
        self.button_yuhun_drop_statistics.setMinimumSize(QSize(100, 0))
        self.button_yuhun_drop_statistics.setMaximumSize(QSize(100, 16777215))
        self.button_mode_single = QRadioButton(self.yuhun_1)
        self.buttonGroup_mode = QButtonGroup(MainWindow)
        self.buttonGroup_mode.setObjectName(u"buttonGroup_mode")
        self.buttonGroup_mode.addButton(self.button_mode_single)
        self.button_mode_single.setObjectName(u"button_mode_single")
        self.button_mode_single.setGeometry(QRect(128, 28, 51, 20))
        self.button_mode_single.setFont(font3)
        self.button_mode_team = QRadioButton(self.yuhun_1)
        self.buttonGroup_mode.addButton(self.button_mode_team)
        self.button_mode_team.setObjectName(u"button_mode_team")
        self.button_mode_team.setGeometry(QRect(78, 28, 51, 20))
        self.button_mode_team.setFont(font3)
        self.button_mode_team.setMouseTracking(True)
        self.label_mode = QLabel(self.yuhun_1)
        self.label_mode.setObjectName(u"label_mode")
        self.label_mode.setGeometry(QRect(20, 28, 48, 21))
        self.label_mode.setFont(font3)
        self.stackedWidget.addWidget(self.yuhun_1)
        self.jiejietupo_2 = QWidget()
        self.jiejietupo_2.setObjectName(u"jiejietupo_2")
        self.button_jiejietupo_3victory = QRadioButton(self.jiejietupo_2)
        self.button_jiejietupo_3victory.setObjectName(u"button_jiejietupo_3victory")
        self.button_jiejietupo_3victory.setGeometry(QRect(80, 60, 51, 20))
        self.button_kaji = QCheckBox(self.jiejietupo_2)
        self.button_kaji.setObjectName(u"button_kaji")
        self.button_kaji.setGeometry(QRect(80, 100, 47, 20))
        self.button_jiejietupo_9victory = QRadioButton(self.jiejietupo_2)
        self.button_jiejietupo_9victory.setObjectName(u"button_jiejietupo_9victory")
        self.button_jiejietupo_9victory.setGeometry(QRect(140, 60, 51, 20))
        self.label_refresh_rule = QLabel(self.jiejietupo_2)
        self.label_refresh_rule.setObjectName(u"label_refresh_rule")
        self.label_refresh_rule.setGeometry(QRect(10, 60, 48, 16))
        self.label_refresh_rule.setFont(font3)
        self.stackedWidget.addWidget(self.jiejietupo_2)
        self.daoguantupo_3 = QWidget()
        self.daoguantupo_3.setObjectName(u"daoguantupo_3")
        self.button_guanzhan = QCheckBox(self.daoguantupo_3)
        self.button_guanzhan.setObjectName(u"button_guanzhan")
        self.button_guanzhan.setGeometry(QRect(60, 40, 51, 20))
        self.stackedWidget.addWidget(self.daoguantupo_3)
        self.groupBox_info = QGroupBox(self.tab_home)
        self.groupBox_info.setObjectName(u"groupBox_info")
        self.groupBox_info.setGeometry(QRect(250, 10, 281, 411))
        self.text_num = QLineEdit(self.groupBox_info)
        self.text_num.setObjectName(u"text_num")
        self.text_num.setEnabled(True)
        self.text_num.setGeometry(QRect(90, 30, 71, 21))
        self.text_num.setFont(font2)
        self.text_num.setMaxLength(32764)
        self.text_print = QTextBrowser(self.groupBox_info)
        self.text_print.setObjectName(u"text_print")
        self.text_print.setGeometry(QRect(10, 60, 261, 341))
        self.text_print.setFont(font)
        self.label_completion = QLabel(self.groupBox_info)
        self.label_completion.setObjectName(u"label_completion")
        self.label_completion.setGeometry(QRect(22, 32, 52, 17))
        self.button_start = QPushButton(self.tab_home)
        self.button_start.setObjectName(u"button_start")
        self.button_start.setGeometry(QRect(70, 370, 81, 31))
        font4 = QFont()
        font4.setFamilies([u"\u5b8b\u4f53"])
        font4.setPointSize(11)
        font4.setBold(True)
        self.button_start.setFont(font4)
        self.tabWidget.addTab(self.tab_home, "")
        self.tab_setting = QWidget()
        self.tab_setting.setObjectName(u"tab_setting")
        self.scrollArea = QScrollArea(self.tab_setting)
        self.scrollArea.setObjectName(u"scrollArea")
        self.scrollArea.setGeometry(QRect(-4, -1, 551, 431))
        self.scrollArea.setMinimumSize(QSize(0, 0))
        self.scrollArea.setWidgetResizable(True)
        self.scrollAreaWidgetContents = QWidget()
        self.scrollAreaWidgetContents.setObjectName(u"scrollAreaWidgetContents")
        self.scrollAreaWidgetContents.setGeometry(QRect(0, 0, 528, 1000))
        self.scrollAreaWidgetContents.setMinimumSize(QSize(0, 1000))
        self.horizontalLayoutWidget = QWidget(self.scrollAreaWidgetContents)
        self.horizontalLayoutWidget.setObjectName(u"horizontalLayoutWidget")
        self.horizontalLayoutWidget.setGeometry(QRect(40, 60, 451, 51))
        self.horizontalLayout_setting_update = QHBoxLayout(self.horizontalLayoutWidget)
        self.horizontalLayout_setting_update.setObjectName(u"horizontalLayout_setting_update")
        self.horizontalLayout_setting_update.setContentsMargins(0, 0, 0, 0)
        self.horizontalSpacer = QSpacerItem(40, 20, QSizePolicy.Expanding, QSizePolicy.Minimum)

        self.horizontalLayout_setting_update.addItem(self.horizontalSpacer)

        self.setting_update_label = QLabel(self.horizontalLayoutWidget)
        self.setting_update_label.setObjectName(u"setting_update_label")

        self.horizontalLayout_setting_update.addWidget(self.setting_update_label)

        self.setting_update_comboBox = QComboBox(self.horizontalLayoutWidget)
        self.setting_update_comboBox.setObjectName(u"setting_update_comboBox")

        self.horizontalLayout_setting_update.addWidget(self.setting_update_comboBox)

        self.horizontalSpacer_3 = QSpacerItem(40, 20, QSizePolicy.Expanding, QSizePolicy.Minimum)

        self.horizontalLayout_setting_update.addItem(self.horizontalSpacer_3)

        self.horizontalLayoutWidget_2 = QWidget(self.scrollAreaWidgetContents)
        self.horizontalLayoutWidget_2.setObjectName(u"horizontalLayoutWidget_2")
        self.horizontalLayoutWidget_2.setGeometry(QRect(40, 110, 451, 51))
        self.horizontalLayout_setting_xuanshangfengyin = QHBoxLayout(self.horizontalLayoutWidget_2)
        self.horizontalLayout_setting_xuanshangfengyin.setObjectName(u"horizontalLayout_setting_xuanshangfengyin")
        self.horizontalLayout_setting_xuanshangfengyin.setContentsMargins(0, 0, 0, 0)
        self.horizontalSpacer_4 = QSpacerItem(40, 20, QSizePolicy.Expanding, QSizePolicy.Minimum)

        self.horizontalLayout_setting_xuanshangfengyin.addItem(self.horizontalSpacer_4)

        self.setting_xuanshangfengyin_label = QLabel(self.horizontalLayoutWidget_2)
        self.setting_xuanshangfengyin_label.setObjectName(u"setting_xuanshangfengyin_label")

        self.horizontalLayout_setting_xuanshangfengyin.addWidget(self.setting_xuanshangfengyin_label)

        self.setting_xuanshangfengyin_comboBox = QComboBox(self.horizontalLayoutWidget_2)
        self.setting_xuanshangfengyin_comboBox.setObjectName(u"setting_xuanshangfengyin_comboBox")

        self.horizontalLayout_setting_xuanshangfengyin.addWidget(self.setting_xuanshangfengyin_comboBox)

        self.horizontalSpacer_6 = QSpacerItem(40, 20, QSizePolicy.Expanding, QSizePolicy.Minimum)

        self.horizontalLayout_setting_xuanshangfengyin.addItem(self.horizontalSpacer_6)

        self.button_update_record = QPushButton(self.scrollAreaWidgetContents)
        self.button_update_record.setObjectName(u"button_update_record")
        self.button_update_record.setGeometry(QRect(310, 340, 75, 24))
        self.label_GitHub_address = QLabel(self.scrollAreaWidgetContents)
        self.label_GitHub_address.setObjectName(u"label_GitHub_address")
        self.label_GitHub_address.setGeometry(QRect(106, 380, 312, 16))
        self.label_GitHub_address.setFocusPolicy(Qt.NoFocus)
        self.label_GitHub_address.setLayoutDirection(Qt.LeftToRight)
        self.label_GitHub_address.setAlignment(Qt.AlignCenter)
        self.button_restart = QPushButton(self.scrollAreaWidgetContents)
        self.button_restart.setObjectName(u"button_restart")
        self.button_restart.setGeometry(QRect(155, 340, 75, 24))
        self.label_tip_setting_restart = QLabel(self.scrollAreaWidgetContents)
        self.label_tip_setting_restart.setObjectName(u"label_tip_setting_restart")
        self.label_tip_setting_restart.setGeometry(QRect(191, 22, 141, 21))
        self.label_tip_setting_restart.setMouseTracking(False)
        self.label_tip_setting_restart.setLayoutDirection(Qt.LeftToRight)
        self.label_tip_setting_restart.setAlignment(Qt.AlignCenter)
        self.scrollArea.setWidget(self.scrollAreaWidgetContents)
        self.tabWidget.addTab(self.tab_setting, "")
        MainWindow.setCentralWidget(self.centralwidget)

        self.retranslateUi(MainWindow)

        self.tabWidget.setCurrentIndex(0)
        self.stackedWidget.setCurrentIndex(0)


        QMetaObject.connectSlotsByName(MainWindow)
    # setupUi

    def retranslateUi(self, MainWindow):
        MainWindow.setWindowTitle(QCoreApplication.translate("MainWindow", u"MainWindow", None))
        self.groupBox_basic.setTitle(QCoreApplication.translate("MainWindow", u"\u57fa\u672c\u529f\u80fd", None))
        self.button_game_handle.setText(QCoreApplication.translate("MainWindow", u"\u6e38\u620f\u68c0\u6d4b", None))
        self.combo_choice.setPlaceholderText(QCoreApplication.translate("MainWindow", u"\u8bf7\u9009\u62e9\u529f\u80fd", None))
        self.label_function.setText(QCoreApplication.translate("MainWindow", u"<html><head/><body><p><span style=\" font-size:10pt;\">\u529f\u80fd</span></p></body></html>", None))
        self.label_numbers.setText(QCoreApplication.translate("MainWindow", u"<html><head/><body><p><span style=\" font-size:10pt;\">\u6b21\u6570</span></p></body></html>", None))
        self.groupBox_senior.setTitle(QCoreApplication.translate("MainWindow", u"\u9ad8\u7ea7\u8bbe\u7f6e", None))
        self.label_driver.setText(QCoreApplication.translate("MainWindow", u"\u662f\u5426\u53f8\u673a", None))
        self.button_driver_False.setText(QCoreApplication.translate("MainWindow", u"\u5426", None))
        self.button_driver_True.setText(QCoreApplication.translate("MainWindow", u"\u662f", None))
        self.button_passengers_3.setText(QCoreApplication.translate("MainWindow", u"3", None))
        self.button_passengers_2.setText(QCoreApplication.translate("MainWindow", u"2", None))
        self.label_passengers.setText(QCoreApplication.translate("MainWindow", u"\u7ec4\u961f\u4eba\u6570", None))
        self.label_tips.setText(QCoreApplication.translate("MainWindow", u"TextLabel", None))
        self.button_yuhun_drop_statistics.setText(QCoreApplication.translate("MainWindow", u"\u6389\u843d\u7edf\u8ba1beta", None))
        self.button_mode_single.setText(QCoreApplication.translate("MainWindow", u"\u5355\u4eba", None))
        self.button_mode_team.setText(QCoreApplication.translate("MainWindow", u"\u7ec4\u961f", None))
        self.label_mode.setText(QCoreApplication.translate("MainWindow", u"\u6a21\u5f0f", None))
        self.button_jiejietupo_3victory.setText(QCoreApplication.translate("MainWindow", u"3\u80dc", None))
        self.button_kaji.setText(QCoreApplication.translate("MainWindow", u"\u5361\u7ea7", None))
        self.button_jiejietupo_9victory.setText(QCoreApplication.translate("MainWindow", u"9\u80dc", None))
        self.label_refresh_rule.setText(QCoreApplication.translate("MainWindow", u"\u5237\u65b0\u89c4\u5219", None))
        self.button_guanzhan.setText(QCoreApplication.translate("MainWindow", u"\u89c2\u6218", None))
        self.groupBox_info.setTitle(QCoreApplication.translate("MainWindow", u"\u4fe1\u606f", None))
        self.text_num.setPlaceholderText(QCoreApplication.translate("MainWindow", u"\u5b8c\u6210\u60c5\u51b5", None))
        self.text_print.setPlaceholderText(QCoreApplication.translate("MainWindow", u"\u8f93\u51fa\u4fe1\u606f", None))
        self.label_completion.setText(QCoreApplication.translate("MainWindow", u"<html><head/><body><p><span style=\" font-size:10pt;\">\u5b8c\u6210\u60c5\u51b5</span></p></body></html>", None))
        self.button_start.setText(QCoreApplication.translate("MainWindow", u"\u5f00\u59cb", None))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.tab_home), QCoreApplication.translate("MainWindow", u"\u9996\u9875", None))
        self.setting_update_label.setText(QCoreApplication.translate("MainWindow", u"\u66f4\u65b0\u65b9\u5f0f", None))
        self.setting_xuanshangfengyin_label.setText(QCoreApplication.translate("MainWindow", u"\u60ac\u8d4f\u5c01\u5370", None))
        self.button_update_record.setText(QCoreApplication.translate("MainWindow", u"\u66f4\u65b0\u8bb0\u5f55", None))
        self.label_GitHub_address.setText(QCoreApplication.translate("MainWindow", u"<html><head/><body><p><span style=\" text-decoration: underline;\">https://github.com/AquamarineCyan/Onmyoji_Python</span></p></body></html>", None))
        self.button_restart.setText(QCoreApplication.translate("MainWindow", u"\u91cd\u542f", None))
        self.label_tip_setting_restart.setText(QCoreApplication.translate("MainWindow", u"\u90e8\u5206\u8bbe\u7f6e\u9700\u8981\u91cd\u542f\u751f\u6548", None))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.tab_setting), QCoreApplication.translate("MainWindow", u"\u8bbe\u7f6e", None))
    # retranslateUi

